service/lib/agama/dbus/storage/interfaces/component.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2024] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "dbus"
+
+module Agama
+  module DBus
+    module Storage
+      module Interfaces
+        # Interface for devices that are used as component of other device (e.g., physical volume,
+        # MD RAID device, etc).
+        #
+        # @note This interface is intended to be included by {Device} if needed.
+        module Component
+          COMPONENT_INTERFACE = "org.opensuse.Agama.Storage1.Component"
+          private_constant :COMPONENT_INTERFACE
+
+          # Type of component.
+          #
+          # @return [String] Possible values:
+          #   "physical_volume"
+          #   "md_device"
+          #   "raid_device"
+          #   "multipath_wire"
+          #   "bcache_device"
+          #   "bcache_cset_device"
+          #   "md_btrfs_device"
+          def component_type
+            types = {
+              lvm_vg:      "physical_volume",
+              md:          "md_device",
+              dm_raid:     "raid_device",
+              multipath:   "multipath_wire",
+              bcache:      "bcache_device",
+              bcache_cset: "bcache_cset_device",
+              btrfs:       "md_btrfs_device"
```

Comment:
do we want to have those string localized or not?

Reply:
No, they are a type of id.